tests/cross/src/main/scala/tests/pc/BaseTypeDefinitionSuite.scala

```diff
@@ -0,0 +1,73 @@
+package tests.pc
+
+import tests.BasePCSuite
+import scala.concurrent.duration.Duration
+import scala.meta.internal.metals.CompilerOffsetParams
+import collection.JavaConverters._
+import org.eclipse.{lsp4j => l}
+
+class BaseTypeDefinitionSuite extends BasePCSuite {
+
+  override def beforeAll(): Unit = {
+    indexJDK()
+    indexScalaLibrary()
+  }
+
+  def locationFromString(str: String): l.Location = {
+    def positionFromString(str: String): l.Position = {
+      val ints = str.split(":").map(_.toInt)
+      new l.Position(ints.head, ints.last)
+    }
+
+    val split = str.split("""/\\""")
+    assertEquals(
+      split.length,
+      3,
+      s"Incorrect location string format, is: $str, " +
+        "should be: |url/\\startLine:startCharacter/\\endLine:endCharacter|"
+    )
+
+    val (uri, rest) = (split.head, split.tail)
+    val range =
+      new l.Range(positionFromString(rest.head), positionFromString(rest.last))
+    new l.Location(uri, range)
+  }
+
+  def locationToString(loc: l.Location): String = {
+    val (uri, start, end) =
+      (loc.getUri, loc.getRange.getStart, loc.getRange.getEnd)
+    s"$uri [${start.getLine}:${start.getCharacter} -> ${end.getLine}:${end.getCharacter}]"
+  }
+
+  def check(name: String)(
+      code: String,
+      expected: String,
+      duration: Duration = Duration("3 min")
+  ): Unit = {
+    val filename = "Main.scala"
+
+    val raw = "package a\n" + code
+    val paramsT = params(raw, filename)
+    val typeDef = pc
+      .typeDefinition(
+        CompilerOffsetParams(
+          filename,
+          paramsT._1,
+          paramsT._2
+        )
+      )
+      .get
+      .asScala
+      .toList
+
+    test(name) {
```

Comment:
Move this to the top of the method, otherwise one failing test case will block the subsequent test cases from starting.

Reply:
Like that?